prow/cmd/tackle/main.go

```diff
@@ -161,6 +175,55 @@ func selectProject(choice string) (string, error) {
 	return choice, nil
 }
 
+// selectZone returns the user-selected zone, defaulting to the current gcloud one.
+func selectZone() (string, error) {
+	const MaxZones = 20
+	var choice string
+
+	def, err := currentZone()
+	if err != nil {
+		return "", fmt.Errorf("get current zone: %v", err)
+	}
+
+	fmt.Printf("Available zones:\n")
+
+	zoneList, err := zones(MaxZones)
+	if err != nil {
+		return "", fmt.Errorf("list zones: %v", err)
+	}
+
+	for _, zone := range zoneList {
+		fmt.Println("  *", zone)
+	}
+
+	if len(zoneList) >= MaxZones {
+		fmt.Println("  ...")
+		fmt.Println("Type the name of any zone, including ones not in this truncated list")
+	}
+
+	if def == "" {
+		// Arbitrarily select the first zone as the `default` if unset
+		def = zoneList[0]
+	}
+
+	fmt.Printf("Select zone [%s]: ", def)
+	fmt.Scanln(&choice)
+
+	// use default zone
+	if choice == "" {
+		return def, nil
+	}
+
+	// is this a zone from the list?
+	for _, z := range zoneList {
```

Comment:
This would prevent specifying a zone that isn't one of the 20 that are listed. It should be safe to omit this check and let the cluster creation command fail if the zone DNE.

Reply:
Good catch! This is definitely an errant check. I will remove and delegate the error throwing to `gcloud` command.

Reply:
@cjwagner - another alternative is to limit the # of results *printed* rather than the # of results *returned*, then `zoneList` can hold all options an be validated against appropriately. 

Let me try to encapsulate this functionality into a helper function as you [suggested](https://github.com/kubernetes/test-infra/pull/13417#issuecomment-510700800) and see if it makes sense.